apps/labs/posts/sympy_documentation.md

```diff
@@ -0,0 +1,358 @@
+  for details.
+
+  This has been deprecated since SymPy version 1.10. It
+  will be removed in a future version of SymPy.
+  ```
+
+  These warning messages give detailed information on what is deprecated, what
+  users can replace their code with, what version the deprecation was added
+  in, and a link to a [documentation
+  page](https://docs.sympy.org/latest/explanation/active-deprecations.html#deprecated-sympy-core-compatibility).
+
+- All activate deprecations are listed in a [new page the
+  documentation](https://docs.sympy.org/latest/explanation/active-deprecations.html).
+  This page gives more details about each deprecation than would be appropriate
+  to put in the deprecation message, including details on why each deprecation
+  was made. The page also gives [helpful
+  information](https://docs.sympy.org/dev/explanation/active-deprecations.html#silencing-sympy-deprecation-warnings)
+  on how to silence deprecation warnings.
+
+## Guide on Writing Custom Functions
+
+SymPy comes with hundreds of mathematical functions built-in. But it also
+comes with a standard functionality for users to define their own custom
+functions. This is achieved by subclassing `sympy.Function` and defining
+various methods to specify the symbolic behavior. For example,
+
+```py
+class log(Function):
+    """
+    Simplified version of sympy.log that supports basic evaluation and
+    differentiation.
+    """
+    @classmethod
+    def eval(cls, x):
+        if x == 1:
+            return 0
+
+    def fdiff(self, argindex=1):
+        return 1/self.args[0]
+```
+
+```py
+>>> x = sympy.Symbol('x')
+>>> log(1)
+0
+>>> log(x).diff(x)
+1/x
+```
+
+SymPy now includes an extensive [how-to guide on defining custom symbolic
+functions](https://docs.sympy.org/latest/guides/custom-functions.html). This
+guide is useful to advanced users, but also this exact same
+method is used to define the functions that are included in
+SymPy itself. So this guide serves as both a guide to advanced end-users as
+well as a guide to SymPy developers looking to define or extend one of the
+functions that comes with SymPy.
+
+## Guide on SymPy Best Practices
+
+SymPy has many pitfalls, both for new users and advanced users. The new guide
+on [best
+practices](https://docs.sympy.org/dev/explanation/best-practices.html) goes
+over some of the best practices that should be applied to avoid these
+pitfalls.
+
+For example, one pitfall that many new SymPy users run into is using strings
+as inputs to SymPy functions, like
+
+```py
+>>> from sympy import expand
+>>> expand("(x**2 + x)/x")
+x + 1
+```
+
+It's much better to define symbolic variables and create expressions directly,
+like
+
+```
+>>> from sympy import symbols
+>>>> x = symbols('x')
+>>> expand((x**2 + x)/x)
+x + 1
+```
+
+The [best practices
+page](https://docs.sympy.org/dev/explanation/best-practices.html#avoid-string-inputs)
```

Comment:
This links to a specific section on the page.

Reply:
Yes, I noticed that. I would still remove it. I would even consider paring back more links from the post because I believe there are a few different places where you link to the same page, or different places on the same page. Links are great but each link also represents a fork in the reader's attention, away from your story, so I would consider them carefully (and/or perhaps create a footnotes section?). 

If you want to keep this link, then I have a different suggestion, which I will add in a separate comment.

Another way to think about this. If I were editing this blog post for print, I would look for ways to bring the point of the link into the article itself. So instead of linking to the specific best practice, I would maybe inline a screenshot or a block quote, excerpting that section right here into the article itself. Which you've kind of already done with the code sample above.